src/on_shutdown.c

```diff
@@ -0,0 +1,182 @@
+#include "on_shutdown.h"
+
+#include <stdlib.h>
+#include <small/rlist.h>
+#include <trivia/util.h>
+#include <fiber.h>
+#include <libev/ev.h>
+
+struct on_shutdown_trigger {
+	/*
+	 * rlist entry
```

Comment:
Please, write more informative comments. It's clear from declaration that it's rlist entry :)
For instance, you can say "Organize triggers into list", or remove comment at all.